File: Tracks/typescript/space-age/space-age.ts
export default class SpaceAge {
    seconds: number;

    constructor(seconds: number) {
        this.seconds = seconds;
    }

    calculateAge = (orbitalPeriodInEarthYears: number) =>
        parseFloat((this.seconds / 31557600 / orbitalPeriodInEarthYears).toFixed(2));

    onEarth = () => this.calculateAge(1);
    onMercury = () => this.calculateAge(0.2408467);
    onVenus = () => this.calculateAge(0.61519726);
    onMars = () => this.calculateAge(1.8808158);
    onJupiter = () => this.calculateAge(11.862615);
    onSaturn = () => this.calculateAge(29.447498);
    onUranus = () => this.calculateAge(84.016846);
    onNeptune = () => this.calculateAge(164.79132);
}

File: Tracks/csharp/space-age/SpaceAge.cs
public class SpaceAge
{
    const int SECONDS_IN_EARTH_YEAR = 31557600;
    private readonly double earthYears;
    public SpaceAge(int seconds) => this.earthYears = (double)seconds / SECONDS_IN_EARTH_YEAR;

    public double OnEarth() => this.earthYears;
    public double OnMercury() => this.earthYears / .2408467;
    public double OnVenus() => this.earthYears / .61519726;
    public double OnMars() => this.earthYears / 1.8808158;
    public double OnJupiter() => this.earthYears / 11.862615;
    public double OnSaturn() => this.earthYears / 29.447498;
    public double OnUranus() => this.earthYears / 84.016846;
    public double OnNeptune() => this.earthYears / 164.79132;
}